extensions/ql-vscode/src/variant-analysis/variant-analysis-manager.ts

```diff
@@ -264,16 +264,16 @@ export class VariantAnalysisManager
     } else {
       await this.setVariantAnalysis(variantAnalysis);
 
-      try {
-        const repoStates = await readRepoStates(
-          this.getRepoStatesStoragePath(variantAnalysis.id),
-        );
-        this.repoStates.set(variantAnalysis.id, repoStates);
-      } catch (e) {
-        // Ignore this error, we simply might not have downloaded anything yet
-        this.repoStates.set(variantAnalysis.id, {});
+      const repoStatesFromDisk = await readRepoStates(
+        this.getRepoStatesStoragePath(variantAnalysis.id),
+      );
+
+      if (repoStatesFromDisk) {
+        this.repoStates.set(variantAnalysis.id, repoStatesFromDisk);
       }
 
+      this.repoStates.set(variantAnalysis.id, {});
```

Comment:
I like moving the catch to be inside `readRepoStates`. Unfortunately rewriting this changed the behaviour so now this will currently always overwrite the repo states with `{}`, which isn't what we want. 

We could put the case for when `repoStatesFromDisk` is undefined into an `else` block. Or we could use `|| {}` to supply this default value.

```suggestion
      this.repoStates.set(variantAnalysis.id, repoStatesFromDisk || {})
```